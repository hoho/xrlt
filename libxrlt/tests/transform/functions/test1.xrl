<?xml version="1.0"?>
<xrl:requestsheet xmlns:xrl="http://xrlt.net/Transform">

    <xrl:variable name="hehehe">hahaha</xrl:variable>

    <xrl:function name="func1">
        <xrl:param name="p3" />
        <xrl:param name="p1" select="111" />
        <xrl:param name="p2">222</xrl:param>

        <xrl:variable name="test" select="concat('AAAA', $p2)"/>

        <a>|<xrl:value-of select="$p1" />|<xrl:value-of select="$p2" />|<xrl:value-of select="$p3" />|</a>
        <a>|<xrl:value-of select="$hehehe" />|<xrl:value-of select="$test" />|</a>
    </xrl:function>

    <xrl:function name="func2">
        <xrl:param name="par1" />
        <xrl:param name="par2" select="42" />

        <a>Third call:</a>
        <xrl:apply name="func1" />

        <a>Fourth call:</a>
        <xrl:apply name="func1">
            <xrl:with-param name="p1" select="$par1" />
            <xrl:with-param name="p2" select="$par2" />
        </xrl:apply>

        <a>Fifth call:</a>
        <xrl:apply name="func1">
            <xrl:with-param name="p1">fuckyeah</xrl:with-param>
            <xrl:with-param name="p3">yo</xrl:with-param>
        </xrl:apply>
    </xrl:function>

    <xrl:response>
        <xrl:variable name="ylyly">
            <a>f</a>
            <b>
                <xrl:include>
                    <xrl:href>hehe</xrl:href>
                    <xrl:type>text</xrl:type>
                </xrl:include>
            </b>
        </xrl:variable>

        <xrl:variable name="alala" select="concat(1, $ylyly, 2)" />

        <xrl:value-of select="$hehehe" />

        <xrl:value-of select="$alala" />

        <a>First call:</a>
        <xrl:apply name="func1">
            <xrl:with-param name="p2">22222</xrl:with-param>
        </xrl:apply>

        <a>Second call:</a>
        <xrl:apply name="func1">
            <xrl:with-param name="p4">alala</xrl:with-param>
            <xrl:with-param name="p3">|<xrl:value-of select="123 + 35" />|</xrl:with-param>
        </xrl:apply>

        <xrl:apply name="func2" />
    </xrl:response>

</xrl:requestsheet>
