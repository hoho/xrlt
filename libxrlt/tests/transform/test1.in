id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:3, type:2, last:0, error: 0, data:{"aaa
id:3, type:2, last:1, error: 0, data:bbb": 123, "ccc": true}
id:0, type:0, last:0, error: 0, data:
id:2, type:2, last:0, error: 0, data:<hello>
id:2, type:2, last:0, error: 0, data:world
id:2, type:2, last:1, error: 0, data:</hello>
id:0, type:0, last:0, error: 0, data:
id:1, type:2, last:0, error: 0, data:there are
id:1, type:2, last:0, error: 0, data: many clouds
id:1, type:2, last:1, error: 0, data: in the sky
id:0, type:0, last:0, error: 0, data:
