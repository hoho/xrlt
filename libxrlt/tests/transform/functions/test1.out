XRLT_STATUS_CHUNK
chunk: hahaha
XRLT_STATUS_SUBREQUEST
sr id: 1
sr method: GET
sr type: TEXT
sr url: hehe
sr query: (null)
sr body: (null)
XRLT_STATUS_WAITING
XRLT_STATUS_WAITING
XRLT_STATUS_CHUNK
chunk: 1fAWESOME2
chunk: First call:
chunk: |111|22222||
chunk: |hahaha|AAAA22222|
chunk: Second call:
chunk: |111|222||158||
chunk: |hahaha|AAAA222|
chunk: Third call:
chunk: |111|222||
chunk: |hahaha|AAAA222|
chunk: Fourth call:
chunk: ||42||
chunk: |hahaha|AAAA42|
chunk: Fifth call:
chunk: |fuckyeah|222|yo|
chunk: |hahaha|AAAA222|
XRLT_STATUS_DONE
